### ISSUE TITLE
[Bug]: DayZ install script doesn't account for mission files - servers boot but don't allow connections

### ISSUE THREAD

**Issue by jamiew0w** [GameServerManagers/LinuxGSM#4601]
### User story

DayZserver should ensure the `mpmissions` folder is created and populated.

### Game

Dayz

### Linux distro

Ubuntu 22.04, Debian 12, RedHat 9

### Command

command: install

### Further information

When installing a DayZ server, it doesn't include the `mpmissions` folder, normally located inside `/home/dayzserver/serverfiles/`. The lack of this directory doesn't stop the server from booting, however it does disallow any players from connecting:

![](https://crowdfile.net/UHg5787c.png)

To fix it manually:

```
mkdir /home/dayzserver/serverfiles/mpmissions
wget https://github.com/BohemiaInteractive/DayZ-Central-Economy/archive/refs/tags/DZ_125.zip
unzip DZ_125.zip; rm DZ_125.zip
mv DayZ-Central-Economy/* /home/dayzserver/serverfiles/mpmissions/
rm -rf DayZ-Central-Economy
./dayzserver restart
```

This isn't a direct failure on LGSM's part though I do think it should account for it as part of creating a new server. The newest version of the mission files can be found at https://github.com/BohemiaInteractive/DayZ-Central-Economy/releases/latest

**Comment by dgibbs64** [comment]
This needs to be a priority fix I think

**Comment by jamiew0w** [comment]
the `mods` & `profiles` directories should probably be created too. the `skeleton` command does create all three so maybe that could be invoked prior to install? 
then all that'd need to be changed is changing the output directory from `skel` to `serverfiles` and jobs done

**Comment by ChrisYatesUK** [comment]
Yeah would be good to have it automatically create & configure this as I've been creating the mod directory / profiles etc manually and its caused me a few issues ( nothing major ) but it is a bit of a pain :)

**Comment by airtonix** [comment]
Mouldy Cheeses!

I wasted a whole weekend with no useful outcome over this and i concluded linuxgsm was a just another abandoned and broken hobby project.

now i've seen this ticket and... glad to know it's just a missing mpmission file/directory.


@jamiew0w or @ChrisYatesUK could you outline some speculative bash commands we would run to get over this hump?

**Comment by anthony-mariotti** [comment]
I was messing around with steamcmd in general outside of LGSM and I noticed that when used my main steam account, which owns the game, the mpmissions folder gets populated.

Using my server account which doesn't own DayZ didn't download the mpmissions. Could be something to consider.

**Comment by Ir0nsh007er** [comment]
not solved ?? and this manualy fix work for me !

**Comment by alexferra** [comment]
@anthony-mariotti is right. the account you use must OWN dayz, otherwise mpmissions does not download!

https://discord.gg/AQHV4Avcem
